ui/desktop/src/App.tsx

```diff
@@ -82,24 +86,68 @@ export default function App() {
 
-    const setupExtensions = async () => {
+    // First quickly check if we have model and provider to set chat view
+    const checkRequiredConfig = async () => {
       try {
-        console.log('Setting up extensions...');
+        console.log('Reading GOOSE_PROVIDER and GOOSE_MODEL from config...');
+        const provider = (await read('GOOSE_PROVIDER', false)) as string;
+        const model = (await read('GOOSE_MODEL', false)) as string;
+
+        if (provider && model) {
+          // We have all needed configuration, set chat view immediately
+          console.log(`Found provider: ${provider}, model: ${model}, setting chat view`);
+          setView('chat');
+
+          // Initialize the system in background
+          initializeSystem(provider, model)
+            .then(() => console.log('System initialization successful'))
+            .catch((error) => {
+              console.error('Error initializing system:', error);
+              setFatalError(`System initialization error: ${error.message || 'Unknown error'}`);
+              setView('welcome');
+            });
+        } else {
+          // Missing configuration, show onboarding
+          console.log('Missing configuration, showing onboarding');
+          if (!provider) console.log('Missing provider');
+          if (!model) console.log('Missing model');
+          setView('welcome');
+        }
+      } catch (error) {
+        console.error('Error checking configuration:', error);
+        setFatalError(`Configuration check error: ${error.message || 'Unknown error'}`);
+        setView('welcome');
+      }
+    };
 
-        // Set the ref immediately to prevent duplicate runs
-        initAttemptedRef.current = true;
-        console.log('Set initAttemptedRef to prevent duplicate runs');
+    // Setup extensions in parallel
+    const setupExtensions = async () => {
+      // Set the ref immediately to prevent duplicate runs
+      initAttemptedRef.current = true;
 
+      let refreshedExtensions: FixedExtensionEntry[] = [];
+      try {
         // Force refresh extensions from the backend to ensure we have the latest
         console.log('Getting extensions from backend...');
-        const refreshedExtensions = await getExtensions(true);
+        refreshedExtensions = await getExtensions(true);
         console.log(`Retrieved ${refreshedExtensions.length} extensions`);
+      } catch (error) {
+        console.log('Error getting extensions list');
+        return; // Exit early if we can't get the extensions list
+      }
+
+      // built-in extensions block -- just adds them to config if missing
+      try {
+        console.log('Setting up built-in extensions...');
 
         if (refreshedExtensions.length === 0) {
           // If we still have no extensions, this is truly a first-time setup
           console.log('First-time setup: Adding all built-in extensions...');
           await initializeBuiltInExtensions(addExtension);
```

Comment:
Conceptual question.

If `refreshedExtensions` comes back the first time you ever boot the app as empty, is it any different/required to have a separate `initializeBuiltInExtensions` function? Could we just use a single `syncBuiltInExtensions` that behaves the same was as `initialize` if there were no extensions to begin with?

(No need to address this in this PR if it distracts - just trying to find opportunities to reduce complexity)